kornia/augmentation/augmentation.py

```diff
@@ -1723,3 +1724,80 @@ def apply_transform(
         self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
         return input + params['noise'].to(input.device) * self.std + self.mean
+
+
+class RandomFisheye(AugmentationBase2D):
+    r"""Add random camera radial distortion.
+
+    Args:
+        center_x (torch.Tensor): Ranges to sample respect to x-coordinate center with shape (2,).
+        center_y (torch.Tensor): Ranges to sample respect to y-coordinate center with shape (2,).
+        gamma (torch.Tensor): Ranges to sample for the gamma values respect to optical center with shape (2,).
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+            input tensor. If ``False`` and the input is a tuple the applied transformation wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        p (float): probability of applying the transformation. Default value is 0.5.
+
+    Examples:
```

Comment:
Not related, but it will be nice if we can print some images whilst build the docs to show what is accepted and expected.

Reply:
let's target this for the next feature release. Wanna go for it with a script to automate this ?